httpx/_client.py

```diff
@@ -397,7 +397,7 @@ def _merge_cookies(
         Merge a cookies argument together with any cookies on the client,
         to create the cookies used for the outgoing request.
         """
-        if cookies or self.cookies:
+        if not (cookies is None or self.cookies is None):
```

Comment:
```suggestion
        if not (cookies is None and self.cookies is None):
```
Then I think we can merge it after changelog added.

Reply:
No, we cannot. `self.cookies` is always not `None`.